cmd/sonobuoy/app/operations/gen.go

```diff
@@ -17,20 +17,52 @@ limitations under the License.
 package operations
 
 import (
-	"errors"
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/heptio/sonobuoy/cmd/sonobuoy/app/utils/mode"
+	"github.com/heptio/sonobuoy/pkg/templates"
 )
 
 // GenConfig are the input options for running
 // TODO: Figure out chained subcommands or how to share input options from RunConfig
 type GenConfig struct {
-	Path string
-	opts RunConfig
+	Path     string
+	ModeName mode.Name
+	Image    string
+}
+
+type templateValues struct {
+	E2EFocus       string
```

Comment:
we'll need skip and provider too.  

Ideally, I was hoping gen/run could use @chuckha's tooling to query the cluster and get the provider settings and fill it in properly for us.

Reply:
Neither of those are used by any of the examples right now. What's skip and provider for?

Reply:
skip is the regex list passed to the image, provider right now is plumbing but will be leveraged a lot more in the not-to-distant future.  

We get asked all the time by folks customizing the focus, and skip vars... 
https://github.com/heptio/kube-conformance/blob/master/Dockerfile#L32